var/spack/repos/builtin/packages/margo/package.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Margo(AutotoolsPackage):
+    """Argobots bindings to the Mercury RPC API"""
+
+    homepage = "https://xgitlab.cels.anl.gov/sds/margo"
+    url      = "https://xgitlab.cels.anl.gov/sds/margo/-/archive/v0.4.3/margo-v0.4.3.tar.gz"
+    git      = "https://xgitlab.cels.anl.gov/sds/margo.git"
+
+    maintainers = ['fbudin69500', 'chuckatkins']
+
+    version('develop', branch='master')
+    version('0.4.3', sha256='61a634d6983bee2ffa06e1e2da4c541cb8f56ddd9dd9f8e04e8044fb38657475')
+
+    variant('shared', default=True,
+            description='Also build shared libraries')
+
+    depends_on('autoconf',  type='build')
+    depends_on('automake',  type='build')
+    depends_on('libtool',   type='build')
+    depends_on('m4',        type='build')
+    depends_on('pkgconfig', type='build')
+
+    depends_on('mercury')
+    depends_on('argobots')
+
+    build_directory = 'spack-build'
+
+    def configure_args(self):
+        spec = self.spec
+        args = []
+
+        if '+shared' in spec:
+            args.append('--enable-shared')
+        else:
+            args.append('--disable-shared')
```

Comment:
This has been a very confusing topic with lots of discussion but still no official policy, but I believe the consensus is that a spack package should generate static **or** shared libraries but not static **and** shared libraries.  Given that, I believe the `shared` variant should not only enable/disable shared libs but also conversely disable/enable static libs.  i.e.
```
if '+shared' in spec:
  args.append('--disable-static', '--enable-shared')
￼else:
  args.append('--enable-static', '--disable-shared')
```

@citibeth am I parsing the discussion around this issue correctly?